tensorflow_io/arrow/kernels/arrow_kernels.cc

```diff
@@ -0,0 +1,254 @@
+  }
+  arrow::Status Read(int64_t nbytes, std::shared_ptr<arrow::Buffer>* out) override {
+    return arrow::Status::NotImplemented("Read (Buffer*)");
+  }
+  arrow::Status GetSize(int64_t* size) override {
+    *size = size_;
+    return arrow::Status::OK();
+  }
+  bool supports_zero_copy() const override {
+    return false;
+  }
+  arrow::Status ReadAt(int64_t position, int64_t nbytes, int64_t* bytes_read, void* out) override {
+    StringPiece result;
+    Status status = file_->Read(position, nbytes, &result, (char*)out);
+    if (!(status.ok() || errors::IsOutOfRange(status))) {
+        return arrow::Status::IOError(status.error_message());
+    }
+    *bytes_read = result.size();
+    return arrow::Status::OK();
+  }
+  arrow::Status ReadAt(int64_t position, int64_t nbytes, std::shared_ptr<arrow::Buffer>* out) override {
+    string buffer;
+    buffer.resize(nbytes);
+    StringPiece result;
+    Status status = file_->Read(position, nbytes, &result, (char*)(&buffer[0]));
+    if (!(status.ok() || errors::IsOutOfRange(status))) {
+        return arrow::Status::IOError(status.error_message());
+    }
+    buffer.resize(result.size());
+    return arrow::Buffer::FromString(buffer, out);
+  }
+private:
+  tensorflow::RandomAccessFile* file_;
+  int64 size_;
+};
+
+class ListFeatherColumnsOp : public OpKernel {
+ public:
+  explicit ListFeatherColumnsOp(OpKernelConstruction* context) : OpKernel(context) {
+    env_ = context->env();
+  }
+
+  void Compute(OpKernelContext* context) override {
+    const Tensor& filename_tensor = context->input(0);
+    const string filename = filename_tensor.scalar<string>()();
+
+    const Tensor& memory_tensor = context->input(1);
+    const string& memory = memory_tensor.scalar<string>()();
+    std::unique_ptr<SizedRandomAccessFile> file(new SizedRandomAccessFile(env_, filename, memory));
+    uint64 size;
+    OP_REQUIRES_OK(context, file->GetFileSize(&size));
+
+    std::shared_ptr<ArrowRandomAccessFile> feather_file(new ArrowRandomAccessFile(file.get(), size));
+
+    std::unique_ptr<arrow::ipc::feather::TableReader> reader;
+    arrow::Status s = arrow::ipc::feather::TableReader::Open(feather_file, &reader);
+    OP_REQUIRES(context, s.ok(), errors::Internal(s.ToString()));
+
+    std::vector<string> columns;
+    std::vector<string> dtypes;
+    std::vector<int64> counts;
+    columns.reserve(reader->num_columns());
+    dtypes.reserve(reader->num_columns());
+    counts.reserve(reader->num_columns());
+    for (int i = 0; i < reader->num_columns(); i++) {
+      std::shared_ptr<arrow::Column> column;
+      s = reader->GetColumn(i, &column);
+      OP_REQUIRES(context, s.ok(), errors::Internal(s.ToString()));
+
+      ::tensorflow::DataType data_type;
+      s = ::arrow::adapters::tensorflow::GetTensorFlowType(column->type(), &data_type);
+      if (!s.ok()) {
+        continue;
+      }
+      string dtype = "";
+      switch (data_type) {
```

Comment:
the Arrow `DataType` has a `ToString` method which I think should give the same output and not require the switch here. WDYT?